openshift_metrics/merge.py

```diff
@@ -26,13 +40,21 @@ def main():
         "--upload-to-s3",
         action="store_true"
     )
+    parser.add_argument(
+        "--ignore-hours",
```

Comment:
Both [openstack](https://github.com/CCI-MOC/openstack-billing-from-db/blob/d31b698906f2777fec1da54b10269691ec610e6a/src/openstack_billing_db/billing.py#L21) and [storage](https://github.com/nerc-project/coldfront-plugin-cloud/blob/ce7c8e1bbd5133cee198900d8f671464a7026554/src/coldfront_plugin_cloud/management/commands/calculate_storage_gb_hours.py#L100) billing take this as a date range rather than a hour range. Can you add an extra option that supports the time in date format too? Otherwise the parameters that we pass for excluded ranges will be different for the openshift billing.

Reply:
@knikolla This actually supports both because I use datetime.fromisoformat to convert it to a datetime object.

```
>>> datetime.fromisoformat("1970-01-01")
datetime.datetime(1970, 1, 1, 0, 0)
>>> datetime.fromisoformat("1970-01-01T00:00:00")
datetime.datetime(1970, 1, 1, 0, 0)
```

So, a user could provide just a date which will assume a time of 00:00:00 or a specific time in addition to the date.

The reason I wrote it to accept a specific time was because @joachimweyl said we want to be able to ignore by the hour and not just a date.

While testing this I realized I should be specific in the timezone when the datetime object is created. Otherwise a naive datetime object is created which tries to use the local system time. So, I think I should set the timezone for the user input to be in UTC.

this is what happens when I use naive datetime object to convert from a datetime -> epoch time -> datetime
```
>>> datetime.utcfromtimestamp(datetime.fromisoformat("1970-01-01T00:00:00").timestamp())
datetime.datetime(1970, 1, 1, 5, 0)
```
(it also incorrectly gets my timezone to be 5 hours behind UTC instead of 4)

So, for this I am just going to update the conversion from string to datetime to be timezone aware and manadate that the user (and ultimately nerc-rates) have the ignore times be in UTC. Is that reasonable?

Reply:
btw, when I set the timezone it works correctly
```
>>> datetime.utcfromtimestamp(datetime.fromisoformat("1970-01-01T00:00:00").replace(tzinfo=UTC).timestamp())
datetime.datetime(1970, 1, 1, 0, 0)
```